$OpenBSD$

Patch in OpenBSD paths. Applied upstream, albeit without ports vars:
https://bugs.pypy.org/issue1531

--- lib_pypy/_tkinter/tklib.py.orig	Thu Aug  1 10:57:52 2013
+++ lib_pypy/_tkinter/tklib.py	Fri Sep 13 16:24:32 2013
@@ -1,6 +1,7 @@
 # C bindings with libtcl and libtk.
 
 from cffi import FFI
+import sys
 
 tkffi = FFI()
 
@@ -102,6 +103,15 @@ int Tcl_DoOneEvent(int flags);
 int Tk_GetNumMainWindows();
 """)
 
+if sys.platform.startswith("openbsd"):
+    incdirs = ['${MODTCL_INCDIR}', '${MODTK_INCDIR}', '${X11BASE}/include']
+    linklibs = ['tk85', 'tcl85']
+    libdirs = ['${LOCALBASE}/lib', '${X11BASE}/lib']
+else:
+    incdirs=['/usr/include/tcl'],
+    linklibs=['tcl', 'tk'],
+    libdirs = []
+
 tklib = tkffi.verify("""
 #include <tcl.h>
 #include <tk.h>
@@ -109,6 +119,7 @@ tklib = tkffi.verify("""
 char *get_tk_version() { return TK_VERSION; }
 char *get_tcl_version() { return TCL_VERSION; }
 """,
-include_dirs=['/usr/include/tcl'],
-libraries=['tcl', 'tk'],
+include_dirs=incdirs,
+libraries=linklibs,
+library_dirs = libdirs
 )
